spec/jobs/open_order_cycle_job_spec.rb

```diff
@@ -77,19 +77,21 @@
       expect(OrderCycles::WebhookService)
         .to receive(:create_webhook_job).with(order_cycle, 'order_cycle.opened', now).once
 
-      # Start two jobs in parallel:
-      threads = [
-        Thread.new { OpenOrderCycleJob.perform_now(order_cycle.id) },
-        Thread.new { OpenOrderCycleJob.perform_now(order_cycle.id) },
-      ]
-
-      # Wait for both to jobs to pause.
-      # This can reveal a race condition.
-      sleep 0.1
-
-      # Resume and complete both jobs:
-      breakpoint.unlock
-      threads.each(&:join)
+      expect{
+        # Start two jobs in parallel:
+        threads = [
+          Thread.new { OpenOrderCycleJob.perform_now(order_cycle.id) },
+          Thread.new { OpenOrderCycleJob.perform_now(order_cycle.id) },
+        ]
+
+        # Wait for both to jobs to pause.
+        # This can reveal a race condition.
+        sleep 0.1
+
+        # Resume and complete both jobs:
+        breakpoint.unlock
+        threads.each(&:join)
+      }.to raise_error ActiveRecord::RecordNotFound
```

Comment:
Hm, I guess that the error is raised during `sleep`? In that case we never unlock the breakpoint and join the threads. There will be an orphan thread hanging around until rspec quits. It wouldn't cause any problems but it's also not how the setup of the test was designed. Or does the error only bubble up when we call `join`?

Reply:
I think that the error is raised in the thread, immediately after `breakpoint.unlock`, when the job is permitted to start running.

But when the thread is `join`ed to the main thread, the unhandled exception bubbles up into the spec. I've updated the spec to clarify this.

Reply:
As an aside: For this test, it doesn't really matter which type of error, or indeed if there even was an error (i could have suppressed the error inside the thread). 
But by specifying `RecordNotFound` it at least confirms what I think is happening.